iroh-p2p/src/rpc.rs

```diff
@@ -87,7 +88,7 @@ impl p2p_server::P2p for P2p {
 
     async fn fetch_provider(
         &self,
-        request: Request<iroh_rpc_types::p2p::Key>,
+        request: Request<ProviderKey>,
```

Comment:
not needed now, but thinking about this interface, we probably want to make this a `Cid`, as that is the only thing we use for providers

Reply:
kk leaving a note